File: backend/app/mock_data.py
"""
Mock data script to populate the database with initial data for development
"""
import datetime
import json
from sqlalchemy.orm import Session

from app.core.database import engine, Base, get_db
from app.models.canteen import Canteen
from app.models.menu_item import MenuItem
from app.models.user import User
from app.models.cart import Cart, CartItem
from app.models.order import Order, OrderItem, OrderStep
from app.models.payment import Merchant
from app.core.database import SessionLocal

def add_mock_users(db: Session):
    """Add mock users to the database"""
    users = [
        User(id="a1b2c3d4-e5f6-7890-abcd-ef1234567890", name="John Doe", email="john@example.com", password="hashedpassword", role="user", favoriteCanteens=[], recentOrders=[]),
        User(id="b2c3d4e5-f6a7-8901-bcde-f12345678901", name="Jane Smith", email="jane@example.com", password="hashedpassword", role="user", favoriteCanteens=[], recentOrders=[]),
    ]
    
    for user in users:
        db_user = db.query(User).filter(User.id == user.id).first()
        if not db_user:
            db.add(user)
    
    db.commit()
    print("✅ Mock users added to database")

def add_mock_canteens(db: Session):
    """Add mock canteens to the database"""
    canteens = [
        Canteen(
            id=1,
            name="Central Canteen",
            image="/placeholder.svg",
            location="Main Building",
            rating=4.2,
            openTime="08:00",
            closeTime="20:00",
            isOpen=True,
            description="A popular spot for diverse meals.",
            phone="040-12345678",
            email="centralcanteen@canteen.com",
            schedule={
                "breakfast": "08:00 AM - 10:00 AM",
                "lunch": "12:00 PM - 02:00 PM",
                "dinner": "07:00 PM - 09:00 PM",
                "regular": "08:00 AM - 09:00 PM",
                "evening": "04:00 PM - 06:00 PM",
                "night": None,
                "weekday": "08:00 AM - 09:00 PM",
                "weekend": "08:00 AM - 10:00 PM",
            },
            tags=["Popular", "Diverse"],
        ),
        Canteen(
            id=2,
            name="Library Cafe",
            image="/placeholder.svg",
            location="Library Building",
            rating=3.8,
            openTime="09:00",
            closeTime="18:00",
            isOpen=False,
            description="Cozy cafe near the library.",
            phone="040-87654321",
            email="librarycafe@canteen.com",
            schedule={
                "breakfast": "09:00 AM - 11:00 AM",
                "lunch": "12:30 PM - 02:30 PM",
                "dinner": None,
                "regular": "09:00 AM - 06:00 PM",
                "evening": "04:00 PM - 05:30 PM",
                "night": None,
                "weekday": "09:00 AM - 06:00 PM",
                "weekend": "10:00 AM - 04:00 PM",
            },
            tags=["Cozy", "Quiet"],
        ),
        Canteen(
            id=3,
            name="Tech Hub Canteen",
            image="/placeholder.svg",
            location="Technology Block",
            rating=4.5,
            openTime="07:30",
            closeTime="22:00",
            isOpen=True,
            description="Techies' favorite spot for quick bites.",
            phone="040-11223344",
            email="techhub@canteen.com",
            schedule={
                "breakfast": "07:30 AM - 09:30 AM",
                "lunch": "12:00 PM - 02:00 PM",
                "dinner": "07:00 PM - 10:00 PM",
                "regular": "07:30 AM - 10:00 PM",
                "evening": "05:00 PM - 07:00 PM",
                "night": None,
                "weekday": "07:30 AM - 10:00 PM",
                "weekend": "08:00 AM - 11:00 PM",
            },
            tags=["Tech", "Quick Bites"],
        ),
    ]
    
    for canteen in canteens:
        db_canteen = db.query(Canteen).filter(Canteen.id == canteen.id).first()
        if not db_canteen:
            db.add(canteen)
    
    db.commit()
    print("✅ Mock canteens added to database")

def add_mock_menu_items(db: Session):
    """Add mock menu items to the database"""
    menu_items = [
        MenuItem(
            id=101,
            canteenId=1,
            canteenName="Faculty Lounge",
            name="Masala Dosa",
            description="Crispy rice crepe filled with spiced potato mixture, served with sambar and chutney",
            price=60,
            category="Breakfast",
            image="https://images.unsplash.com/photo-1589301760014-d929f3979dbc?q=80&w=2070&auto=format&fit=crop",
            tags=["South Indian", "Vegetarian"],
            rating=4.5,
            ratingCount=120,
            isAvailable=True,
            preparationTime=15,
            isPopular=True,
            customizationOptions={
                "sizes": [
                    {"name": "small", "price": 50},
                    {"name": "medium", "price": 60},
                    {"name": "large", "price": 70},
                ],
                "additions": [
                    {"name": "Extra Chutney", "price": 10},
                    {"name": "Ghee Roast", "price": 15},
                ],
                "removals": ["Onions", "Green Chilies"],
            },
        ),
        MenuItem(
            id=102,
            canteenId=2,
            canteenName="Faculty Lounge",
            name="Chole Bhature",
            description="Spicy chickpea curry served with deep-fried bread",
            price=80,
            category="Lunch",
            image="https://images.unsplash.com/photo-1589352911312-5d218efc96be?q=80&w=1974&auto=format&fit=crop",
            tags=["North Indian", "Vegetarian"],
            rating=4.3,
            ratingCount=95,
            isAvailable=True,
            preparationTime=20,
            isPopular=True,
            customizationOptions={
                "sizes": [
                    {"name": "small", "price": 70},
                    {"name": "medium", "price": 80},
                    {"name": "large", "price": 90},
                ],
                "additions": [
                    {"name": "Extra Bhature", "price": 20},
                    {"name": "Onions on Side", "price": 0},
                ],
                "removals": ["Spices"],
            },
        ),
        MenuItem(
            id=103,
            canteenId=1,
            canteenName="Central Canteen",
            name="Chicken Biryani",
            description="Fragrant basmati rice cooked with chicken, spices, and herbs",
            price=120,
            category="Lunch",
            image="https://images.unsplash.com/photo-1589309736404-be8c25f8dea8?q=80&w=1974&auto=format&fit=crop",
            tags=["Hyderabadi", "Non-Vegetarian"],
            rating=4.7,
            ratingCount=150,
            isAvailable=True,
            preparationTime=30,
            isPopular=True,
            customizationOptions={
                "sizes": [
                    {"name": "small", "price": 100},
                    {"name": "medium", "price": 120},
                    {"name": "large", "price": 140},
                ],
                "additions": [
                    {"name": "Extra Raita", "price": 15},
                    {"name": "Extra Spicy", "price": 0},
                ],
                "removals": ["Coriander"],
            },
        ),
        MenuItem(
            id=104,
            canteenId=1,
            canteenName="Central Canteen",
            name="Veg Pulao",
            description="Basmati rice cooked with mixed vegetables and mild spices",
            price=90,
            category="Dinner",
            image="https://images.unsplash.com/photo-1596797038530-2c107aa4606c?q=80&w=1935&auto=format&fit=crop",
            tags=["North Indian", "Vegetarian"],
            rating=4.0,
            ratingCount=80,
            isAvailable=False,
            preparationTime=25,
            isPopular=False,
            customizationOptions={
                "sizes": [
                    {"name": "small", "price": 80},
                    {"name": "medium", "price": 90},
                    {"name": "large", "price": 100},
                ],
                "additions": [{"name": "Extra Raita", "price": 15}],
                "removals": ["Peas"],
            },
        ),
    ]
    
    for item in menu_items:
        db_item = db.query(MenuItem).filter(MenuItem.id == item.id).first()
        if not db_item:
            db.add(item)
    
    db.commit()
    print("✅ Mock menu items added to database")

def add_mock_cart_data(db: Session):
    """Add mock cart data to the database"""
    # Create cart for user 1
    cart1 = db.query(Cart).filter(Cart.userId == "a1b2c3d4-e5f6-7890-abcd-ef1234567890").first()
    if not cart1:
        cart1 = Cart(
            userId="a1b2c3d4-e5f6-7890-abcd-ef1234567890",
            items=[
                {
                    "menuItemId": 1,
                    "quantity": 1,
                    "selectedSize": "Medium",
                    "selectedExtras": json.dumps(["Extra Paneer"]),
                    "specialInstructions": "Extra spicy",
                    "location": "Table 5",
                    "name": "Masala Dosa",
                    "price": 60.0,
                    "canteenId": 1,
                    "canteenName": "Central Canteen",
                    "customizations": json.dumps({
                        "size": "Medium",
                        "additions": ["Extra Paneer"],
                        "removals": ["Onions"],
                        "notes": "Extra spicy"
                    })
                },
                {
                    "menuItemId": 3,
                    "quantity": 2,
                    "selectedSize": "Large",
                    "selectedExtras": json.dumps(["Less Sugar", "Regular Ice"]),
                    "specialInstructions": "Extra cold",
                    "location": "Table 7",
                    "name": "Chicken Biryani",
                    "price": 120.0,
                    "canteenId": 1,
                    "canteenName": "Central Canteen",
                    "customizations": json.dumps({
                        "size": "Large",
                        "additions": ["Less Sugar", "Regular Ice"],
                        "removals": [],
                        "notes": "Extra cold"
                    })
                }
            ],
            createdAt=datetime.datetime.utcnow().isoformat(),
            updatedAt=datetime.datetime.utcnow().isoformat()
        )
        db.add(cart1)
    
    # Create cart for user 2
    cart2 = db.query(Cart).filter(Cart.userId == "b2c3d4e5-f6a7-8901-bcde-f12345678901").first()
    if not cart2:
        cart2 = Cart(
            userId="b2c3d4e5-f6a7-8901-bcde-f12345678901",
            items=[
                {
                    "menuItemId": 2,
                    "quantity": 2,
                    "selectedSize": "Small",
                    "selectedExtras": json.dumps(["Extra Chutney"]),
                    "specialInstructions": "Extra crispy",
                    "location": "Table 3",
                    "name": "Chole Bhature",
                    "price": 80.0,
                    "canteenId": 2,
                    "canteenName": "Library Cafe",
                    "customizations": json.dumps({
                        "size": "Small",
                        "additions": ["Extra Chutney"],
                        "removals": [],
                        "notes": "Extra crispy"
                    })
                },
                {
                    "menuItemId": 5,
                    "quantity": 1,
                    "selectedSize": "Medium",
                    "selectedExtras": json.dumps(["Extra Salt", "Ketchup"]),
                    "specialInstructions": "Extra crispy",
                    "location": "Table 1",
                    "name": "Veg Pulao",
                    "price": 90.0,
                    "canteenId": 2,
                    "canteenName": "Library Cafe",
                    "customizations": json.dumps({
                        "size": "Medium",
                        "additions": ["Extra Salt", "Ketchup"],
                        "removals": [],
                        "notes": "Extra crispy"
                    })
                }
            ],
            createdAt=datetime.datetime.utcnow().isoformat(),
            updatedAt=datetime.datetime.utcnow().isoformat()
        )
        db.add(cart2)
    
    db.commit()
    print("✅ Mock cart data added to database")

def add_mock_orders(db: Session):
    """Add mock orders to the database"""
    # Clear existing order data
    db.query(OrderStep).delete()
    db.query(OrderItem).delete()
    db.query(Order).delete()
    
    # Active order for current date (April 17, 2025)
    active_order = Order(
        userId="a1b2c3d4-e5f6-7890-abcd-ef1234567890",
        canteenId=1,
        totalAmount=450.00,
        status="Processing",
        orderTime=datetime.datetime.strptime("2025-04-16T12:15:00.000Z", "%Y-%m-%dT%H:%M:%S.%fZ").isoformat(),
        paymentMethod="Card",
        paymentStatus="Paid",
        phone="1234567890",
        pickupTime="12:45 PM",
        isPreOrder=False
    )
    db.add(active_order)
    db.flush()

    # Order steps for active order
    order_steps = [
        OrderStep(
            orderId=active_order.id,
            status="Order Placed",
            description="Your order has been received by the vendor.",
            time="12:15 PM",
            completed=1,
            current=0
        ),
        OrderStep(
            orderId=active_order.id,
            status="Preparing",
            description="The kitchen is preparing your food.",
            time="12:20 PM",
            completed=0,
            current=1
        ),
        OrderStep(
            orderId=active_order.id,
            status="Ready for Pickup",
            description="Your order is ready for pickup.",
            time="",
            completed=0,
            current=0
        ),
        OrderStep(
            orderId=active_order.id,
            status="Completed",
            description="Your order has been picked up.",
            time="",
            completed=0,
            current=0
        )
    ]
    for step in order_steps:
        db.add(step)
        
    # Order items for active order
    order_items = [
        OrderItem(
            orderId=active_order.id,
            itemId=101,
            quantity=1,
            customizations=json.dumps(["Extra Butter", "Medium Spicy"]),
            note="Make it extra creamy"
        ),
        OrderItem(
            orderId=active_order.id,
            itemId=102,
            quantity=2,
            customizations=json.dumps([]),
            note=""
        ),
        OrderItem(
            orderId=active_order.id,
            itemId=103,
            quantity=1,
            customizations=json.dumps(["Large Portion"]),
            note=""
        )
    ]
    for item in order_items:
        db.add(item)
    
    # Order history
    order_history1 = Order(
        userId="a1b2c3d4-e5f6-7890-abcd-ef1234567890",
        canteenId=1,
        totalAmount=320.00,
        status="Completed",
        orderTime=datetime.datetime.strptime("2025-04-12T15:30:00.000Z", "%Y-%m-%dT%H:%M:%S.%fZ").isoformat(),
        paymentMethod="Card",
        paymentStatus="Paid",
        phone="1234567890",
        isPreOrder=False
    )
    db.add(order_history1)
    db.flush()
    
    order_history2 = Order(
        userId="a1b2c3d4-e5f6-7890-abcd-ef1234567890",
        canteenId=2,
        totalAmount=280.00,
        status="Completed",
        orderTime=datetime.datetime.strptime("2025-04-10T13:15:00.000Z", "%Y-%m-%dT%H:%M:%S.%fZ").isoformat(),
        paymentMethod="Card",
        paymentStatus="Paid",
        phone="1234567890",
        isPreOrder=False
    )
    db.add(order_history2)
    db.flush()
    
    order_history3 = Order(
        userId="a1b2c3d4-e5f6-7890-abcd-ef1234567890",
        canteenId=3,
        totalAmount=520.00,
        status="Cancelled",
        orderTime=datetime.datetime.strptime("2025-04-05T19:30:00.000Z", "%Y-%m-%dT%H:%M:%S.%fZ").isoformat(),
        cancelledTime=datetime.datetime.strptime("2025-04-05T19:40:00.000Z", "%Y-%m-%dT%H:%M:%S.%fZ").isoformat(),
        cancellationReason="Changed my mind",
        paymentMethod="Cash",
        paymentStatus="Refunded",
        phone="1234567890",
        isPreOrder=False
    )
    db.add(order_history3)
    db.flush()
    
    # Add items for first history order
    history_items1 = [
        OrderItem(
            orderId=order_history1.id,
            itemId=101,
            quantity=1,
            customizations=json.dumps(["Extra Spicy", "Regular Portion"]),
            note=""
        ),
        OrderItem(
            orderId=order_history1.id,
            itemId=102,
            quantity=2,
            customizations=json.dumps([]),
            note=""
        ),
        OrderItem(
            orderId=order_history1.id,
            itemId=103,
            quantity=1,
            customizations=json.dumps([]),
            note=""
        )
    ]
    
    # Add items for second history order
    history_items2 = [
        OrderItem(
            orderId=order_history2.id,
            itemId=104,
            quantity=1,
            customizations=json.dumps(["No Onions", "Extra Cheese"]),
            note=""
        ),
        OrderItem(
            orderId=order_history2.id,
            itemId=101,
            quantity=1,
            customizations=json.dumps(["Extra Salt"]),
            note=""
        ),
        OrderItem(
            orderId=order_history2.id,
            itemId=103,
            quantity=1,
            customizations=json.dumps([]),
            note=""
        )
    ]
    
    # Add items for third history order
    history_items3 = [
        OrderItem(
            orderId=order_history3.id,
            itemId=101,
            quantity=1,
            customizations=json.dumps(["Extra Veggies"]),
            note=""
        ),
        OrderItem(
            orderId=order_history3.id,
            itemId=102,
            quantity=1,
            customizations=json.dumps(["Dry", "Extra Spicy"]),
            note=""
        ),
        OrderItem(
            orderId=order_history3.id,
            itemId=103,
            quantity=1,
            customizations=json.dumps([]),
            note=""
        ),
        OrderItem(
            orderId=order_history3.id,
            itemId=104,
            quantity=1,
            customizations=json.dumps(["No Eggs"]),
            note=""
        )
    ]
    
    # Add all history order items
    for item in history_items1 + history_items2 + history_items3:
        db.add(item)
    
    db.commit()
    print("✅ Mock orders added to database")

#  Payment mock merchant data

def create_merchant_data(db: Session):
    db = SessionLocal()
    try:
        # Check if merchants already exist
        existing_merchant = db.query(Merchant).first()
        if existing_merchant:
            print("Merchant data already exists")
            return

        # Create merchants for each canteen
        merchants = [
            Merchant(
                canteen_id=1,
                name="VC",
                razorpay_merchant_id="north_merchant",
                razorpay_key_id="rzp_test_LG7CwFwevu2LNK",
                razorpay_key_secret="xAJyi89fMjP5IRGxua2hsNCE",
                is_active=True
            ),
            Merchant(
                canteen_id=2,
                name="Tantra",
                razorpay_merchant_id="south_merchant",
                razorpay_key_id="rzp_test_LG7CwFwevu2LNK",
                razorpay_key_secret="xAJyi89fMjP5IRGxua2hsNCE",
                is_active=True
            ),
            Merchant(
                canteen_id=3,
                name="David's",
                razorpay_merchant_id="cafeteria_merchant",
                razorpay_key_id="rzp_test_LG7CwFwevu2LNK",
                razorpay_key_secret="xAJyi89fMjP5IRGxua2hsNCE",
                is_active=True
            )
        ]

        db.add_all(merchants)
        db.commit()
        print("Created merchant data")
    except Exception as e:
        db.rollback()
        print(f"Error creating merchant data: {e}")
    finally:
        db.close()



def initialize_mock_data():
    """Initialize database with mock data"""
    try:
        # Create tables if they don't exist
        Base.metadata.create_all(bind=engine)
        print("✅ Database tables created")
        
        # Add mock data
        db = next(get_db())
        add_mock_users(db)
        add_mock_canteens(db)
        add_mock_menu_items(db)
        add_mock_cart_data(db)
        add_mock_orders(db)
        create_merchant_data(db)

        
        
        print("✅ All mock data added successfully")
    except Exception as e:
        print(f"❌ Error initializing mock data: {e}")

if __name__ == "__main__":
    initialize_mock_data()

File: frontend/src/data/mockData.ts
// Mock data for the Smart Canteen application

// Canteens
export const canteens = [
  {
    id: 1,
    name: "Himalaya Food Court",
    image: "/placeholder.svg",
    location: "Near Academic Block",
    rating: 4.5,
    openTime: "08:00",
    closeTime: "22:00",
    isOpen: true,
    description: "Multi-cuisine restaurant offering a variety of dishes from across India.",
    phone: "040-23456789"
  },
  {
    id: 2,
    name: "Vindhya Canteen",
    image: "/placeholder.svg",
    location: "Vindhya Building Ground Floor",
    rating: 4.2,
    openTime: "09:00",
    closeTime: "21:00",
    isOpen: true,
    description: "Fast food and quick bites for students on the go.",
    phone: "040-23456788"
  },
  {
    id: 3,
    name: "Faculty Dining",
    image: "/placeholder.svg",
    location: "Admin Block",
    rating: 4.8,
    openTime: "08:30",
    closeTime: "20:30",
    isOpen: true,
    description: "Premium dining experience with table service and gourmet options.",
    phone: "040-23456787"
  },
  {
    id: 4,
    name: "South Campus Cafeteria",
    image: "/placeholder.svg",
    location: "South Campus Main Building",
    rating: 3.9,
    openTime: "08:00",
    closeTime: "21:00",
    isOpen: false,
    description: "Budget-friendly meals with a focus on South Indian cuisine.",
    phone: "040-23456786"
  },
  {
    id: 5,
    name: "Night Canteen",
    image: "/placeholder.svg",
    location: "Hostel Complex",
    rating: 4.0,
    openTime: "18:00",
    closeTime: "03:00",
    isOpen: true,
    description: "Late night food options for hostel students. Popular for midnight snacks.",
    phone: "040-23456785"
  }
];

// Food Categories
export const categories = [
  { id: 1, name: "Breakfast", icon: "coffee" },
  { id: 2, name: "Lunch", icon: "utensils" },
  { id: 3, name: "Dinner", icon: "moon" },
  { id: 4, name: "Snacks", icon: "cookie" },
  { id: 5, name: "Beverages", icon: "mug-hot" },
  { id: 6, name: "Desserts", icon: "ice-cream" }
];

// Food Items
export const menuItems = [
  {
    id: "101",
    canteenId: "1",
    canteenName: "Faculty Lounge",
    name: "Masala Dosa",
    description: "Crispy rice crepe filled with spiced potato mixture, served with sambar and chutney",
    price: 60,
    category: "Breakfast",
    image: "https://images.unsplash.com/photo-1589301760014-d929f3979dbc?q=80&w=2070&auto=format&fit=crop",
    tags: ["South Indian", "Vegetarian"],
    rating: 4.5,
    ratingCount: 120,
    isAvailable: true,
    preparationTime: 15,
    isPopular: true,
    customizationOptions: {
      sizes: [
        { name: "small" as const, price: 50 },
        { name: "medium" as const, price: 60 },
        { name: "large" as const, price: 70 },
      ],
      additions: [
        { name: "Extra Chutney", price: 10 },
        { name: "Ghee Roast", price: 15 },
      ],
      removals: ["Onions", "Green Chilies"],
    },
  },
  {
    id: "102",
    canteenId: "2",
    canteenName: "Faculty Lounge",
    name: "Chole Bhature",
    description: "Spicy chickpea curry served with deep-fried bread",
    price: 80,
    category: "Lunch",
    image: "https://images.unsplash.com/photo-1589352911312-5d218efc96be?q=80&w=1974&auto=format&fit=crop",
    tags: ["North Indian", "Vegetarian"],
    rating: 4.3,
    ratingCount: 95,
    isAvailable: true,
    preparationTime: 20,
    isPopular: true,
    customizationOptions: {
      sizes: [
        { name: "small" as const, price: 70 },
        { name: "medium" as const, price: 80 },
        { name: "large" as const, price: 90 },
      ],
      additions: [
        { name: "Extra Bhature", price: 20 },
        { name: "Onions on Side", price: 0 },
      ],
      removals: ["Spices"],
    },
  },
  {
    id: "103",
    canteenId: "1",
    canteenName: "Central Canteen",
    name: "Chicken Biryani",
    description: "Fragrant basmati rice cooked with chicken, spices, and herbs",
    price: 120,
    category: "Lunch",
    image: "https://images.unsplash.com/photo-1589309736404-be8c25f8dea8?q=80&w=1974&auto=format&fit=crop",
    tags: ["Hyderabadi", "Non-Vegetarian"],
    rating: 4.7,
    ratingCount: 150,
    isAvailable: true,
    preparationTime: 30,
    isPopular: true,
    customizationOptions: {
      sizes: [
        { name: "small" as const, price: 100 },
        { name: "medium" as const, price: 120 },
        { name: "large" as const, price: 140 },
      ],
      additions: [
        { name: "Extra Raita", price: 15 },
        { name: "Extra Spicy", price: 0 },
      ],
      removals: ["Coriander"],
    },
  },
  {
    id: "104",
    canteenId: "1",
    canteenName: "Central Canteen",
    name: "Veg Pulao",
    description: "Basmati rice cooked with mixed vegetables and mild spices",
    price: 90,
    category: "Dinner",
    image: "https://images.unsplash.com/photo-1596797038530-2c107aa4606c?q=80&w=1935&auto=format&fit=crop",
    tags: ["North Indian", "Vegetarian"],
    rating: 4.0,
    ratingCount: 80,
    isAvailable: false,
    preparationTime: 25,
    isPopular: false,
    customizationOptions: {
      sizes: [
        { name: "small" as const, price: 80 },
        { name: "medium" as const, price: 90 },
        { name: "large" as const, price: 100 },
      ],
      additions: [{ name: "Extra Raita", price: 15 }],
      removals: ["Peas"],
    },
  },
];

// Orders
export const orders = [
  {
    id: 1001,
    userId: 101,
    canteenId: 1,
    items: [
      { 
        itemId: 1, 
        quantity: 2, 
        customizations: ["Extra Cheese", "No Onion"],
        note: "Make it spicy please" 
      },
      { 
        itemId: 3, 
        quantity: 1, 
        customizations: [],
        note: "" 
      }
    ],
    totalAmount: 300,
    status: "delivered",
    orderTime: "2025-04-17T12:30:00",
    confirmedTime: "2025-04-17T12:32:00",
    preparingTime: "2025-04-17T12:35:00",
    readyTime: "2025-04-17T12:45:00",
    deliveryTime: "2025-04-17T12:50:00",
    paymentMethod: "UPI",
    paymentStatus: "Paid",
    customerNote: "Please ensure the food is hot when delivered.",
    discount: 0,
    phone: "9876543210",
    pickupTime: null,
    isPreOrder: false,
    cancelledTime: null,
    cancellationReason: null
  },
  {
    id: 1002,
    userId: 101,
    canteenId: 2,
    items: [
      { 
        itemId: 5, 
        quantity: 1, 
        customizations: ["No Cheese"],
        note: "Extra napkins please" 
      }
    ],
    totalAmount: 180,
    status: "ready",
    orderTime: "2025-04-18T10:15:00",
    confirmedTime: "2025-04-18T10:17:00",
    preparingTime: "2025-04-18T10:20:00",
    readyTime: "2025-04-18T10:30:00",
    deliveryTime: null,
    paymentMethod: "Wallet",
    paymentStatus: "Paid",
    customerNote: "",
    discount: 0,
    phone: "9876543210",
    pickupTime: null,
    isPreOrder: false,
    cancelledTime: null,
    cancellationReason: null
  },
  {
    id: 1003,
    userId: 102,
    canteenId: 3,
    items: [
      { 
        itemId: 8, 
        quantity: 2, 
        customizations: [],
        note: "" 
      },
      { 
        itemId: 9, 
        quantity: 1, 
        customizations: ["Iced"],
        note: "Less sugar" 
      }
    ],
    totalAmount: 250,
    status: "preparing",
    orderTime: "2025-04-18T11:00:00",
    confirmedTime: "2025-04-18T11:02:00",
    preparingTime: "2025-04-18T11:05:00",
    readyTime: null,
    deliveryTime: null,
    paymentMethod: "Credit Card",
    paymentStatus: "Paid",
    customerNote: "",
    discount: 0,
    phone: "9876543210",
    pickupTime: null,
    isPreOrder: false,
    cancelledTime: null,
    cancellationReason: null
  },
  {
    id: 1004,
    userId: 103,
    canteenId: 1,
    items: [
      { 
        itemId: 2, 
        quantity: 1, 
        customizations: ["Extra Spicy"],
        note: "" 
      }
    ],
    totalAmount: 150,
    status: "cancelled",
    orderTime: "2025-04-17T15:30:00",
    confirmedTime: null,
    preparingTime: null,
    readyTime: null,
    deliveryTime: null,
    paymentMethod: "UPI",
    paymentStatus: "Refunded",
    customerNote: "",
    discount: 0,
    phone: "9876543210",
    pickupTime: null,
    isPreOrder: false,
    cancelledTime: "2025-04-17T15:35:00",
    cancellationReason: "Item out of stock"
  },
  {
    id: 1005,
    userId: 101,
    canteenId: 2,
    items: [
      { 
        itemId: 5, 
        quantity: 2, 
        customizations: [],
        note: "" 
      },
      { 
        itemId: 6, 
        quantity: 1, 
        customizations: ["Extra Sauce"],
        note: "" 
      }
    ],
    totalAmount: 400,
    status: "confirmed",
    orderTime: "2025-04-18T09:45:00",
    confirmedTime: "2025-04-18T09:47:00",
    preparingTime: null,
    readyTime: null,
    deliveryTime: null,
    paymentMethod: "Debit Card",
    paymentStatus: "Paid",
    customerNote: "Please pack separately",
    discount: 40,
    phone: "9876543210",
    pickupTime: null,
    isPreOrder: false,
    cancelledTime: null,
    cancellationReason: null
  },
  {
    id: 1006,
    userId: 102,
    canteenId: 3,
    items: [
      { 
        itemId: 8, 
        quantity: 1, 
        customizations: [],
        note: "" 
      }
    ],
    totalAmount: 130,
    status: "pending",
    orderTime: "2025-04-18T11:30:00",
    confirmedTime: null,
    preparingTime: null,
    readyTime: null,
    deliveryTime: null,
    paymentMethod: "Cash",
    paymentStatus: "Pending",
    customerNote: "",
    discount: 0,
    phone: "9876543210",
    pickupTime: null,
    isPreOrder: false,
    cancelledTime: null,
    cancellationReason: null
  },
  {
    id: 1007,
    userId: 101,
    canteenId: 1,
    items: [
      { 
        itemId: 1, 
        quantity: 3, 
        customizations: ["Extra Cheese"],
        note: "Birthday celebration" 
      },
      { 
        itemId: 4, 
        quantity: 2, 
        customizations: [],
        note: "" 
      }
    ],
    totalAmount: 500,
    status: "confirmed",
    orderTime: "2025-04-19T18:00:00",
    confirmedTime: "2025-04-19T18:02:00",
    preparingTime: null,
    readyTime: null,
    deliveryTime: null,
    paymentMethod: "UPI",
    paymentStatus: "Paid",
    customerNote: "Special celebration, please add candles",
    discount: 50,
    phone: "9876543210",
    pickupTime: "2025-04-19T19:00:00",
    isPreOrder: true,
    cancelledTime: null,
    cancellationReason: null
  }
];

// User data
export const userData = {
  id: 1,
  name: "Aryan Kumar",
  email: "aryan.kumar@example.edu",
  role: "student",
  favoriteCanteens: [1, 2],
  recentOrders: [1001, 1002]
};

// Payment method options
export const paymentMethodOptions = [
  { id: 1, name: "Credit Card" },
  { id: 2, name: "Debit Card" },
  { id: 3, name: "UPI" },
  { id: 4, name: "Wallet" },
  { id: 5, name: "Cash" },
  { id: 6, name: "Pay Later" },
];

// Payment status options
export const paymentStatusOptions = [
  { id: 1, name: "Pending" },
  { id: 2, name: "Paid" },
  { id: 3, name: "Failed" },
  { id: 4, name: "Refunded" },
];

// Order status options
export const orderStatusOptions = [
  { id: 1, name: "pending", label: "Pending", color: "bg-yellow-500" },
  { id: 2, name: "confirmed", label: "Confirmed", color: "bg-blue-500" },
  { id: 3, name: "preparing", label: "Preparing", color: "bg-blue-500" },
  { id: 4, name: "ready", label: "Ready", color: "bg-green-500" },
  { id: 5, name: "delivered", label: "Delivered", color: "bg-green-500" },
  { id: 6, name: "cancelled", label: "Cancelled", color: "bg-red-500" },
];
